### PR TITLE
BIOWDL-331: Setup Centrifuge pipeline.

### PR DESCRIPTION
**Pull request by JasperBoom**
Krona plot still needs to be added. Will be in later PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,20 @@
-# metagenomics
\ No newline at end of file
+# sequence-classification
+
+This repository contains the [Biowdl](https://github.com/biowdl) pipeline usable for processing metagenomics data sets using [Centrifuge](https://github.com/infphilo/sequence-classification).
```
Comment (DavyCats) [inline README.md]: that link to centrifuge doesn't work

File: docs/index.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+---
+layout: default
+title: Home
+---
+
```
Comment (DavyCats) [inline docs/index.md:5]: Please add a small description of the pipeline

File: docs/index.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+This pipeline is part of [BioWDL](https://biowdl.github.io/)
+developed by the SASC team at [Leiden University Medical Center](https://www.lumc.nl/).
+
+## Usage
+You can run the pipeline using
+[Cromwell](http://cromwell.readthedocs.io/en/stable/):
+
```
Comment (DavyCats) [inline docs/index.md:15]: Please add a link to the inputs overview (and add the inputs overview)

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+```bash
+java -jar cromwell-<version>.jar run -i inputs.json pipeline.wdl
```
Comment (DavyCats) [inline docs/index.md]: The wdl file could probably use a more descriptive name than `pipeline.wdl`

File: docs/index.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+### Inputs
+Inputs are provided through a JSON file. The minimally required inputs are
+described below, but additional inputs are available.
+A template containing all possible inputs can be generated using
+Womtool as described in the
+[WOMtool documentation](http://cromwell.readthedocs.io/en/stable/WOMtool/).
+
+```json
+{
+}
```
Comment (DavyCats) [inline docs/index.md:34]: I'm guessing there are some inputs required

File: requirements-test.txt [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# These are the programs used for testing this biowdl workflow
+# These requirements can be installed with conda with the bioconda channel activated
+# For more information on how to set up conda with bioconda channel see:
+# http://bioconda.github.io/#install-conda
+# this file can be installed with "conda install --file requirements-test.txt"
+cromwell
+pytest-workflow
+miniwdl
```
Comment (DavyCats) [inline requirements-test.txt:8]: add wdl-aid

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+      - TEST=pairedEnd
+      - _JAVA_OPTIONS="-Dconfig.file=tests/cromwell.conf"
+    - env:
+      - TEST=singleEnd
+      - _JAVA_OPTIONS="-Dconfig.file=tests/cromwell.conf"
+    # Below is for testing if all files are valid and if all submodules are up to date.
+    - env: TEST="Womtool validate and submodule up to date"
+      stage: lint
+      install:
+        - conda install cromwell miniwdl
```
Comment (DavyCats) [inline .travis.yml]: This should also install wdl-aid

**Review by DavyCats — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (JasperBoom) [reply README.md]: Good catch, thank you

**Review by rhpvorderman — CHANGES_REQUESTED** [review]
Looks good! Two things:
1. Can you gzip the fastq files?
2. Is it possible to use the first read of the read pair as a single-end? This will cut down the needed test files from 6 to 4.

**Comment by JasperBoom** [comment]
Yes and yes, I have done both now.

File: docs/index.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+java -jar cromwell-<version>.jar run -i inputs.json sequence-classification.wdl
+```
+
+### Inputs
+Inputs are provided through a JSON file. The minimally required inputs are
+described below, but additional inputs are available.
+A template containing all possible inputs can be generated using
+Womtool as described in the
+[WOMtool documentation](http://cromwell.readthedocs.io/en/stable/WOMtool/).
+
```
Comment (DavyCats) [inline docs/index.md:27]: Please add a link to the inputs overview.

File: docs/index.md [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+### Output
+The workflow will output trimmed reads from the QC pipeline, a centrifuge
+classification file, a alignment metrics file and a report file per sample.
+
+## Contact
+<p>
+  <!-- Obscure e-mail address for spammers -->
+For any questions about running this pipeline and feature request (such as
+adding additional tools and options), please use the
+<a href='https://github.com/biowdl/TALON-WDL/issues'>github issue tracker</a>
```
Comment (DavyCats) [inline docs/index.md]: This shouldn't ben linking to the TALON pipeline

**Review by DavyCats — CHANGES_REQUESTED** [review]

File: docs/index.md [hunk 10]
```diff
@@ -0,0 +1,123 @@
+---
+layout: default
+title: Home
+---
+
+This pipeline uses Centrifuge for novel microbial classification and enables rapid,
```
Comment (rhpvorderman) [inline docs/index.md]: Novel? Centrifuge is also quite old already :wink:.

File: tests/data/norovirus.mutations.vcf [hunk 11]
```diff
@@ -0,0 +1,157 @@
+##fileformat=VCFv4.1
```
Comment (rhpvorderman) [inline tests/data/norovirus.mutations.vcf]: Is this file important test data? It does not matter for the classification, does it?

File: tests/data/norovirus.mutations.txt [hunk 12]
```diff
@@ -0,0 +1,137 @@
+NC_001959.2	422	A	G	3
```
Comment (rhpvorderman) [inline tests/data/norovirus.mutations.txt]: Is this file important test data? It does not matter for the classification, does it?

File: tests/functional/README.md [hunk 13]
```diff
@@ -0,0 +1,3 @@
+# Functional tests
```
Comment (rhpvorderman) [inline tests/functional/README.md]: Since there are no functional tests, this folder can be deleted.

File: docs/index.md [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+<sampleId>|<libId>|<rgId>|<Path to first FastQ file.>|<MD5 checksum string.>||
+<sampleId>|<libId>|<rgId>|<Path to first FastQ file.>|<MD5 checksum string.>||
+
+NOTE: R1_md5, R2 and R2_md5 are optional do not have to be filled. And additional fields may be added (eg. for documentation purposes), these will be ignored by the pipeline.
+
+After creating the table in a spreadsheet program it can be saved in 
+csv format.
+
+##### YAML format
+The sample configuration can also be a YML file which adheres to the following
```
Comment (rhpvorderman) [inline docs/index.md]: This YAML format is legacy. Can this be removed?

File: tests/data/README.md [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+  * GCF_007609015.1
+  * GCF_007609035.1
+  * GCF_008703965.1
+  * GCF_008703985.1
+  * GCF_008704005.1
+  * GCF_008704025.1
+  * GCF_008711635.1
+
+* Generate test data:
+  * [dwgsim](https://github.com/nh13/DWGSIM) reference_sequences.fna norovirus
```
Comment (rhpvorderman) [inline tests/data/README.md]: Did you set a seed value? If not, then the test data is not reproducible.

File: tests/samplesheets/pairedEnd.yml [hunk 16]
```diff
@@ -0,0 +1,21 @@
+samples:
```
Comment (rhpvorderman) [inline tests/samplesheets/pairedEnd.yml]: These yaml files can be removed I think.
It is a legacy format, and it is not that easy to work with in my opinion. If you think they are easier than the CSV files you can keep them in.

**Review by rhpvorderman — CHANGES_REQUESTED** [review]

File: docs/index.md (hunk 10 above)
Comment (JasperBoom) [reply docs/index.md]: I took that line from the Centrifuge documentation :stuck_out_tongue_winking_eye:. But I can change it of course.

File: tests/data/norovirus.mutations.vcf (hunk 11 above)
Comment (JasperBoom) [reply tests/data/norovirus.mutations.vcf]: It is not, will remove it.

File: tests/data/norovirus.mutations.txt (hunk 12 above)
Comment (JasperBoom) [reply tests/data/norovirus.mutations.txt]: It is not, will remove it.

**Review by rhpvorderman — APPROVED** [review]

**Review by DavyCats — APPROVED** [review]

**Pull request merged**